src/sentry/tasks/organization_mapping.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from datetime import datetime, timedelta
+
+from django.contrib.postgres.aggregates import ArrayAgg
+from django.db.models import Count
+from django.utils import timezone
+
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.services.hybrid_cloud.organization import organization_service
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics
+from sentry.utils.query import RangeQuerySetWrapper
+
+ORGANIZATION_MAPPING_EXPIRY = timedelta(hours=2)
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.tasks.organization_mapping.repair_mappings",
+    queue="hybrid_cloud.control_repair",
+    default_retry_delay=5,
+    max_retries=5,
+)  # type: ignore
+def repair_mappings() -> None:
+    metrics.incr("sentry.hybrid_cloud.tasks.organizationmapping.start", sample_rate=1.0)
+    with metrics.timer("sentry.hybrid_cloud.tasks.organizationmapping.repair", sample_rate=1.0):
+        expiration_threshold_time = timezone.now() - ORGANIZATION_MAPPING_EXPIRY
+        _verify_mappings(expiration_threshold_time)
+        _remove_duplicate_mappings(expiration_threshold_time)
+
+
+def _verify_mappings(expiration_threshold_time: datetime) -> None:
+    # For each unverified mapping in the control silo
+    for mapping in RangeQuerySetWrapper(OrganizationMapping.objects.filter(verified=False)):
+        # Lookup organization record in the region silo
+        org = organization_service.get_organization_by_id(
+            id=mapping.organization_id, slug=mapping.slug
```

Comment:
For the control silo implementation of this service how would we resolve which region the organization is expected to exist in? Should this query also include the region or would the control silo implementation of this service method use the mapping table to locate the region?

Reply:
The latter. There's a good example of how this could work in Zach's [RPC PR](https://github.com/getsentry/sentry/pull/43219)

Reply:
Ok. Using the organization id to do routing via the mapping table makes sense.